app/models/user.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  extend FriendlyId
+  friendly_id :name, use: :slugged
+
+  before_save { self.email = email.downcase }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
+
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :email, presence: true, length: { maximum: 255 },
+                    format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+
+  has_many :posts, class_name: 'Post', foreign_key: 'user_id', dependent: :destroy
```

Comment:
Do you really need the class_name and the foreign_key? I advise you go through the docs again if not sure.